src/libstd/collections/hash/table.rs

```diff
@@ -395,12 +421,14 @@ impl<K, V, M: Deref<Target=RawTable<K, V>>> FullBucket<K, V, M> {
     }
 }
 
-impl<K, V, M: Deref<Target=RawTable<K, V>> + DerefMut> FullBucket<K, V, M> {
+// We don't need a `Take` trait currently. This is why a mutable reference
+// to the table is required.
```

Comment:
This comment doesn't really make sense to me. Maybe something that explicitly references `stash` and the purpose of restricting to mutable references, like "We take a mutable reference to the table instead of accepting anything that implements `DerefMut` to prevent `take` from being called on `stash`ed buckets."?